neurom/features/morphology.py

```diff
@@ -163,6 +174,8 @@ def elevation(neurite):
 @feature(shape=(...,))
 def trunk_vectors(morph, neurite_type=NeuriteType.all):
     """Calculate the vectors between all the trunks of the morphology and the soma center."""
+    _assert_soma_center(morph, "trunk_vectors")
```

Comment:
Instead of increasing the logic inside the function, the soma check could be a decorator, checking that morph has a soma and raising using the function's name if not. Would that work?

Reply:
Yeah I thought about something like this but the `sholl_crossings` feature can actually work with empty soma if `center` and `radii` are provided. So a decorator would not work in all cases. But we could create a decorator for other features and just something specific for this one.

Reply:
Is it however a realistic scenario that the feature can work without a soma? Couldn't we make it strict even if it isn't always needed?

Reply:
I think it is since it just counts the number of intersections between neurites and a set of circles. The centers of these circles are usually the center of the soma but I guess it could be anything, though I don't really see the point of such feature :smile: 
And in fact, this function was not a feature until `Neurom == 3`, it was just an internal function. Now that it is a feature, I think it should not accept a `center` parameter any more and it should always use the soma center. WDYT?

Reply:
(ofc if we change the signature we will have to target the v4 branch)

Reply:
No, you are right. The sholl crossings can also take a list of neurites. It is fine the way you have it.

Reply:
ok